enhancements/api-review/add-ImagePolicy-to-config.openshift.io-v1alpha1.md

```diff
@@ -0,0 +1,336 @@
+// For more information about the format, see the document
+// about the location field:
+// https://github.com/containers/image/blob/main/
+// docs/containers-policy.json.5.md#docker
+type Image struct {
+	Hostname string `json:"hostname,omitempty"`
+}
+
+type Policy struct {
+	// OIDCIssuer contains the expected OIDC issuer.
+	// Requires SubjectEmail to be set and KeyData to be empty.
+	// Example: "https://expected.OIDC.issuer/"
+	OIDCIssuer string `json:"oidcIssuer,omitempty"`
+
+	// SubjectEmail holds the email address of the subject.
+	// Example: "expected-signing-user@example.com"
+	// Requires OIDCIssuer to be set and KeyData to be empty.
+	SubjectEmail string `json:"subjectEmail,omitempty"`
+
+	// KeyData contains inline base64 data of the public key.
+	// Can be empty if the image got signed keyless.
+	// Requires OIDCIssuer and SubjectEmail to be empty.
+	KeyData string `json:"keyData,omitempty"`
+
+	// SignedIdentity specifies what image identity the signature
+	// claims about the image.
+	SignedIdentity Identity `json:"signedIdentity,omitempty"`
+
+	// FulcioCAData contains inline base64 data for the fulcio
+	// CA certificate.
+	// Defaults to the base64 encoded contents of:
+	// https://raw.githubusercontent.com/sigstore/root-signing/main/
+	// targets/fulcio_v1.crt.pem
+	// QUESTION: Can we omit it in favor of the upstream default?
+	FulcioCAData string `json:"fulcioCAData,omitempty"`
+
+	// RekorKeyData contains inline base64 data of the rekor public key.
+	// Defaults to the base64 encoded contents of:
+	// https://raw.githubusercontent.com/sigstore/root-signing/main/
+	// targets/rekor.pub
+	// QUESTION: Can we omit it in favor of the upstream default?
+	RekorKeyData string `json:"rekorKeyData,omitempty"`
+}
+
+type Identity struct {
+	// Defaults to "matchRepository"
+	IdentityMatchPolicy IdentityMatchPolicy `json:"identityMatchPolicy,…"`
+	Prefix              string              `json:"prefix,omitempty"`
+	SignedPrefix        string              `json:"signedPrefix,omitempty"`
+	DockerReference     string              `json:"dockerReference,omitempty"`
+	DockerRepository    string              `json:"dockerRepository,omitempty"`
+}
+
+type IdentityMatchPolicy string
+
+const (
+	IdentityMatchPolicyMatchRepository        = "matchRepository"
+	IdentityMatchPolicyMatchExact             = "matchExact"
+	IdentityMatchPolicyRemapIdentity          = "remapIdentity"
+	IdentityMatchPolicyMatchRepoDigestOrExact = "matchRepoDigestOrExact"
+	IdentityMatchPolicyExactRepository   = "exactRepository"
+	IdentityMatchPolicyExactReference    = "exactReference"
+)
+
+type ImagePolicyStatus struct {
+	// PolicyJSON contains the whole policy applied to the namespace
+	// which got written to disk. This includes cluster-wide policies
+	// from the `openshift-config` namespace as well.
+	PolicyJSON string `json:"policyJSON,omitempty"`
+
+	// QUESTION: DO we need conditions?
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
```

Comment:
Conditions I would define:
- **Pending**: Resource exists but waiting to be created
- **Creating**: Resource is currently created
- **Available**: Everything is up to date (on every node)
- **Updating**: Resource is being updated
- **Deleting**: Resource is being deleted

Reply:
would define these to the openshfit/api.